src/VisualStudio/Core/Test/PullMemberUp/PullMemberUpViewModelTest.vb

```diff
@@ -0,0 +1,336 @@
+            Dim markUp = <Text><![CDATA[
+interface Level2Interface
+{
+}
+                             
+interface Level1Interface : Level2Interface
+{
+}
+
+class Level1BaseClass: Level2Interface
+{
+}
+
+class MyClass : Level1BaseClass, Level1Interface
+{
+    public void G$$oo()
+    {
+    }
+
+    public double e => 2.717;
+
+    private double pi => 3.1416;
+}"]]></Text>
+            Dim viewModel = Await GetViewModelAsync(markUp, LanguageNames.CSharp)
+            viewModel.DeSelectAllMembers()
+
+            Assert.True(Not viewModel.SelectAllCheckBoxState)
+            Assert.True(Not viewModel.ThreeStateEnable)
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsPullMemberUp)>
+        Public Async Function TestPullMemberUp_SelectInterfaceDisableFieldCheckbox() As Task
+            Dim markUp = <Text><![CDATA[
+interface Level2Interface
+{
+}
+                             
+interface Level1Interface : Level2Interface
+{
+}
+
+class Level1BaseClass: Level2Interface
+{
+}
+
+class MyClass : Level1BaseClass, Level1Interface
+{
+    public void G$$oo()
+    {
+    }
+
+    public double e => 2.717;
+
+    public const days = 365;
+
+    private double pi => 3.1416;
+
+    protected float goldenRadio = 0.618;
+
+    internal float gravitational = 6.67e-11;
+}"]]></Text>
+            Dim viewModel = Await GetViewModelAsync(markUp, LanguageNames.CSharp)
+            Dim baseTypeTree = viewModel.Destinations()
+
+            Assert.Equal("Level1Interface", baseTypeTree(0).MemberName)
+            viewModel.SelectedDestination = baseTypeTree(0)
+
+            For Each member In viewModel.Members.Where(Function(memberViewModel) memberViewModel.MemberSymbol.IsKind(SymbolKind.Field))
+                Assert.False(member.IsCheckable)
+            Next
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsPullMemberUp)>
+        Public Async Function TestPullMemberUp_SelectClassEnbaleFieldCheckbox() As Task
+            Dim markUp = <Text><![CDATA[
+interface Level2Interface
+{
+}
+                             
+interface Level1Interface : Level2Interface
+{
+}
+
+class Level1BaseClass: Level2Interface
+{
+}
+
+class MyClass : Level1BaseClass, Level1Interface
+{
+    public void G$$oo()
+    {
+    }
+
+    public double e => 2.717;
+
+    public const days = 365;
+
+    private double pi => 3.1416;
+
+    protected float goldenRadio = 0.618;
+
+    internal float gravitational = 6.67e-11;
+}"]]></Text>
+            Dim viewModel = Await GetViewModelAsync(markUp, LanguageNames.CSharp)
+            Dim baseTypeTree = viewModel.Destinations()
+
+            Assert.Equal("Level1Interface", baseTypeTree(0).MemberName)
+            viewModel.SelectedDestination = baseTypeTree(0)
+
+            Assert.Equal("Level1BaseClass", baseTypeTree(1).MemberName)
+            viewModel.SelectedDestination = baseTypeTree(1)
```

Comment:
Why setting this multiple times? Is this trying to test what happens if you change it multiple times in a session, or just copy/paste duplication?

Reply:
No, it is simulating the scenario when user first select interface (the field check box is disabled), then select class (field check box will be resumed), in this test, baseTypeTree(0) is an interface (The assert is just to make sure it is an interface), same as the baseTypeTree(1) below.